sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/degrade/DegradeRule.java

```diff
@@ -206,16 +205,14 @@ public boolean passCheck(Context context, DefaultNode node, int acquireCount, Ob
             }
         }
 
-        synchronized (lock) {
-            if (!cut) {
-                // Automatically degrade.
-                cut = true;
+        if (!cut.get()) {
```

Comment:
It's unnecessary to check here as the CAS will compare the old value.

Reply:
I think if this property is true, then there is no need to perform CAS operations because CAS also affects performance.